Add support for Python 3.8

<!--- Provide a general summary of the issue in the Title above -->
## Context
AWS Lambda now supports Python 3.8 but Zappa does not.  Supporting Python 3.8 is potentially a big deal since it uses Amazon Linux 2, which includes some C libraries that are important for some Python packages.
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- Also, please make sure that you are running Zappa _from a virtual environment_ and are using Python 2.7/3.6 -->

## Expected Behavior
Zappa should be able to deploy using `runtime: "python3.8"`
<!--- Tell us what should happen -->

## Actual Behavior
When running `zappa deploy staging`:

```
[1574908829969] [ERROR] RuntimeError: This version of Python (3.8) is not supported!
Zappa (and AWS Lambda) support the following versions of Python: ['2.7', '3.6', '3.7']
Traceback (most recent call last):
  File "/var/lang/lib/python3.8/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/var/lang/lib/python3.8/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/var/task/handler.py", line 22, in <module>
    from zappa.middleware import ZappaWSGIMiddleware
  File "/var/task/zappa/__init__.py", line 12, in <module>
    raise RuntimeError(err_msg)
```
<!--- Tell us what happens instead -->

## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
Possibly an easy fix is just updating the **__init__.py** file.  Snippet below:

```
import sys

SUPPORTED_VERSIONS = [(2, 7), (3, 6), (3, 7)]

python_major_version = sys.version_info[0]
python_minor_version = sys.version_info[1]

if (python_major_version, python_minor_version) not in SUPPORTED_VERSIONS:
    formatted_supported_versions = ['{}.{}'.format(mav, miv) for mav, miv in SUPPORTED_VERSIONS]
    err_msg = 'This version of Python ({}.{}) is not supported!\n'.format(python_major_version, python_minor_version) +\
              'Zappa (and AWS Lambda) support the following versions of Python: {}'.format(formatted_supported_versions)
    raise RuntimeError(err_msg)

__version__ = '0.48.2'
```


## Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug include code to reproduce, if relevant -->
1. `zappa init`.
2. Set the `runtime` variables to `"python3.8" in the **zappa_settings.yml** file.
3. `zappa deploy`

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Zappa version used: `zappa==0.48.2`
* Operating System and Python version: Mac OSX 10.13.6.  Python 3.7.4
* The output of `pip freeze`:
```
area==1.1.1
argcomplete==1.9.3
atomicwrites==1.3.0
attrs==19.3.0
awslogs==0.11.0
boto3==1.9.251
botocore==1.12.251
certifi==2019.9.11
cfn-flip==1.2.1
chardet==3.0.4
Click==7.0
clickclick==1.2.2
connexion==2.4.0
coverage==4.5.4
docutils==0.15.2
durationpy==0.5
Flask==1.1.1
Flask-Cors==3.0.8
future==0.16.0
hjson==3.0.1
idna==2.8
importlib-metadata==0.23
inflection==0.3.1
itsdangerous==1.1.0
Jinja2==2.10.3
jmespath==0.9.3
jsonschema==2.6.0
kappa==0.6.0
lambda-packages==0.20.0
MarkupSafe==1.1.1
mgrs==1.3.5
more-itertools==7.2.0
openapi-spec-validator==0.2.8
packaging==19.2
placebo==0.9.0
pluggy==0.13.0
py==1.8.0
pyparsing==2.4.2
pytest==5.2.1
pytest-cov==2.8.1
python-dateutil==2.6.1
python-slugify==1.2.4
PyYAML==5.1.2
requests==2.22.0
s3transfer==0.2.1
sentry-sdk==0.12.3
Shapely==1.6.4.post2
short-url==1.2.2
six==1.12.0
sqlparse==0.3.0
swagger-ui-bundle==0.0.5
termcolor==1.1.0
toml==0.10.0
tqdm==4.19.1
troposphere==2.5.2
Unidecode==1.1.1
urllib3==1.25.6
wcwidth==0.1.7
Werkzeug==0.16.0
wsgi-request-logger==0.4.6
zappa==0.48.2
zipp==0.6.0
```

* Link to your project (optional):
* Your `zappa_settings.py`: 

```
---

features:
  app_function: app.api_app
  # aws_region: us-west-2
  profile_name: xxxx
  s3_bucket: zappa-bazze-events
  exclude:
    - "*.gz"
    - "*.pyc"
    - "*.zip"
    - ".tox"
    - "__pycache__"
    - "tests"
    - ".venv"
    - ".cache"
    - ".pip-cache"
    - ".vscode"
  project_name: events
  domain: example.com
  certificate_arn: arn:aws:acm:us-east-1:294182326609:certificate/889ef603-ed89-433a-af5e-20379c8ff874
  timeout_seconds: 120
  api_key_required: false
  environment_variables:
    APP_MODE: feature
    SENTRY_DSN: ''
  cors: true
  timeout_seconds: 300
  manage_roles: false
  role_name: events-production-ZappaLambdaExecutionRole
  # extra_permissions: {"Effect":"Allow","Action":["athena:*"],"Resource":["*"]}

staging:
  extends: features
  domain: example.com
  environment_variables:
    APP_MODE: staging
    SENTRY_DSN: ''
  runtime: "python3.8"

production:
  extends: features
  domain: example.com
  environment_variables:
    APP_MODE: production
    SENTRY_DSN: ''
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,6 +3,7 @@ python:
   - "2.7"
   - "3.6"
   - "3.7"
+  - "3.8"
 dist: xenial
 # command to install dependencies
 cache:

--- a/setup.py
+++ b/setup.py
@@ -52,6 +52,7 @@ setup(
         'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
         'Framework :: Django',
         'Framework :: Django :: 1.11',
         'Framework :: Django :: 2.0',

--- a/tests/test_app.py
+++ b/tests/test_app.py
@@ -1,5 +1,13 @@
-from cgi import parse_qs, escape
 from zappa.asynchronous import task
+try:
+    from urllib.parse import parse_qs
+except ImportError:
+    from cgi import parse_qs
+
+try:
+    from html import escape
+except ImportError:
+    from cgi import escape
 
 
 def hello_world(environ, start_response):

--- a/tests/tests.py
+++ b/tests/tests.py
@@ -174,6 +174,19 @@ class TestZappa(unittest.TestCase):
             self.assertTrue(os.path.isfile(path))
             os.remove(path)
 
+    def test_get_manylinux_python38(self):
+        z = Zappa(runtime='python3.8')
+        self.assertIsNotNone(z.get_cached_manylinux_wheel('psycopg2', '2.7.6'))
+        self.assertIsNone(z.get_cached_manylinux_wheel('derp_no_such_thing', '0.0'))
+
+        # mock with a known manylinux wheel package so that code for downloading them gets invoked
+        mock_installed_packages = {'psycopg2': '2.7.6'}
+        with mock.patch('zappa.core.Zappa.get_installed_packages', return_value=mock_installed_packages):
+            z = Zappa(runtime='python3.8')
+            path = z.create_lambda_zip(handler_file=os.path.realpath(__file__))
+            self.assertTrue(os.path.isfile(path))
+            os.remove(path)
+
     def test_should_use_lambda_packages(self):
         z = Zappa(runtime='python2.7')
 

--- a/zappa/__init__.py
+++ b/zappa/__init__.py
@@ -1,6 +1,6 @@
 import sys
 
-SUPPORTED_VERSIONS = [(2, 7), (3, 6), (3, 7)]
+SUPPORTED_VERSIONS = [(2, 7), (3, 6), (3, 7), (3, 8)]
 
 python_major_version = sys.version_info[0]
 python_minor_version = sys.version_info[1]

--- a/zappa/core.py
+++ b/zappa/core.py
@@ -665,7 +665,7 @@ class Zappa(object):
                 # This is a special case!
                 # SQLite3 is part of the _system_ Python, not a package. Still, it lives in `lambda-packages`.
                 # Everybody on Python3 gets it!
-                if self.runtime in ("python3.6", "python3.7"):
+                if self.runtime in ("python3.6", "python3.7", "python3.8"):
                     print(" - sqlite==python3: Using precompiled lambda package")
                     self.extract_lambda_package('sqlite3', temp_project_path)
 

--- a/zappa/utilities.py
+++ b/zappa/utilities.py
@@ -181,8 +181,10 @@ def get_runtime_from_python_version():
     else:
         if sys.version_info[1] <= 6:
             return 'python3.6'
-        else:
+        elif sys.version_info[1] <= 7:
             return 'python3.7'
+        else:
+            return 'python3.8'
 
 ##
 # Async Tasks

